File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/FrmAgregarNuevoProveedor.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports CapaNegocio

Public Partial Class FrmAgregarNuevoProveedor
	Inherits Form
	Private _owner As Proveedores

	Public Sub New()
		InitializeComponent()
	End Sub

	Public Sub New(owner As Proveedores)
		_owner = owner
		InitializeComponent()
	End Sub

	Private Sub MensajeError(mensaje As String)
		MessageBox.Show(mensaje, "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
	End Sub

	Private Sub buttonCancelar_Click(sender As Object, e As EventArgs)
		Me.Close()
	End Sub

	Private Sub buttonGuardar_Click(sender As Object, e As EventArgs)
		Dim mensaje As [String] = NProveedores.Insertar(Me.textBoxNombreCliente.Text, Me.textBoxNombreContacto.Text, Me.textBoxDireccion.Text, Me.textBoxCiudad.Text, Me.textBoxRegion.Text, Me.textBoxPais.Text, _
			Me.textBoxTelefono.Text, Me.textBoxFax.Text, Me.textBoxEmail.Text)

		If mensaje = "Y" Then
			Me._owner.Mensaje([String].Format("El Proveedor {0} ha sido AGREGADO", Me.textBoxNombreCliente.Text))
			Me._owner.Refrescar()
			Me.Close()
		Else
			MensajeError(mensaje)
		End If
	End Sub

End Class

File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/FrmAgregarNuevoProveedor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using CapaNegocio;


namespace CapaPresentacion
{
    public partial class FrmAgregarNuevoProveedor : Form
    {
        Proveedores _owner;
        public FrmAgregarNuevoProveedor()
        {
            InitializeComponent();
        }

        public FrmAgregarNuevoProveedor(Proveedores owner)
        {
            _owner = owner;
            InitializeComponent();
        }

        private void MensajeError(string mensaje)
        {
            MessageBox.Show(mensaje, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
        }

        private void pictureBox2_Click(object sender, EventArgs e)
        {
            String mensaje = NProveedores.Insertar(this.textBoxNombreCliente.Text, this.textBoxDireccion.Text,
                this.textBoxNIT.Text, this.textBoxTelefono.Text, this.textBoxEmail.Text);

            if (mensaje == "Y")
            {
                this._owner.Mensaje(String.Format("El Proveedor {0} ha sido AGREGADO",
                    this.textBoxNombreCliente.Text));
                this._owner.Refrescar();
                this.Close();
            }
            else
            {
                MensajeError(mensaje);
            }
        }

        private void pictureBox3_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void panel1_Paint(object sender, PaintEventArgs e)
        {

        }
    }
}
